piq/tv.py

```diff
@@ -108,14 +107,13 @@ def forward(self, prediction: torch.Tensor) -> torch.Tensor:
     def compute_metric(self, prediction: torch.Tensor) -> torch.Tensor:
         score = total_variation(
             prediction,
-            size_average=False,
-            reduction_type=self.reduction_type
+            reduction=self.reduction,
+            norm_type=self.norm_type,
         )
 
-        if self.reduction == 'mean':
-            score = torch.mean(score, dim=0)
-        elif self.reduction == 'sum':
-            score = torch.sum(score, dim=0)
-        elif self.reduction != 'none':
-            raise ValueError(f'Expected "none"|"mean"|"sum" reduction, got {self.reduction}')
-        return score
+        if self.reduction == 'none':
+            return score
+
+        return {'mean': score.mean,
+                'sum': score.sum
+                }[self.reduction](dim=0)
```

Comment:
I suggest to move it to `forward`. The function returns reduced values already, therefore the processing of the reduction is not needed here.

Reply:
Ready 👍